src/SportsHub.Infrastructure/Repositories/TeamLogoRepository.cs

```diff
@@ -41,7 +41,15 @@ public async Task<bool> DoesTeamLogoAlreadyExistByTeamIdAsync(Guid teamId)
 
         public async Task EditTeamLogoAsync(TeamLogo teamLogo)
         {
-            _context.TeamLogos.Update(teamLogo);
+            var oldTeamLogo = await _context.TeamLogos.FirstOrDefaultAsync(oldTeamLogo => oldTeamLogo.TeamId == teamLogo.TeamId);
```

Comment:
Old sounds wrong in this context :D
```suggestion
            var existingTeamLogo = await _context.TeamLogos.FirstOrDefaultAsync(oldTeamLogo => oldTeamLogo.TeamId == teamLogo.TeamId);
```